include/fc/io/cfile.hpp

```diff
@@ -49,6 +51,11 @@ class cfile {
    ///         "ab+" - open for binary update - create if does not exist
    ///         "rb+" - open for binary update - file must exist
    void open( const char* mode ) {
+      if(is_open()) {
+         fclose(_file.get());
+         _open = false;
+      }
+
       _file.reset( FC_FOPEN( _file_path.generic_string().c_str(), mode ) );
```

Comment:
`_file` is closed twice here. Once above and again with `reset`.

Reply:
thanks for pointing this out.
it seems that using specifying the delete process for the unique pointer on a FILE, might not be the best of ideas.
Maybe just removing the delete process of the unique pointer might be the better idea. I'll make a PR to show you what i mean-